lib/oli/delivery/sections/updates.ex

```diff
@@ -454,6 +454,18 @@ defmodule Oli.Delivery.Sections.Updates do
     end
   end
 
+  defp all_resource_mappings(section) do
+    # Query the section project publications to get all publication ids, then create the
+    # published resource map for each and MERGE them all into one map
+    from(spp in Sections.SectionsProjectsPublications,
+      where: spp.section_id == ^section.id,
+      select: spp.publication_id
+    )
+    |> Repo.all()
+    |> Enum.map(fn publication_id -> MinimalHierarchy.published_resources_map(publication_id) end)
```

Comment:
I noticed this map is making multiple DB requests for each iteration, but I suspect this is okay given the complexity of any alternative solution and since this code executes in an update background job